File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/react_web/src/services/playerService.js
import api from './api';

// GET /players/me/profile
export const getMyProfile = async () => {
  const { data } = await api.get('/players/me/profile');
  return data;
};

// GET /clubs/:clubId/teams/:teamId/players
export const getTeamPlayers = async (clubId, teamId) => {
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/players`);
  return data;
};

// GET /clubs/:clubId/teams/:teamId/players/:playerUserId/profile
export const getPlayerProfile = async (clubId, teamId, playerUserId) => {
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/players/${playerUserId}/profile`);
  return data;
};

// POST /clubs/:clubId/teams/:teamId/players/:playerUserId/profile
export const upsertPlayerProfile = async (clubId, teamId, playerUserId, profileData) => {
  const { data } = await api.post(`/clubs/${clubId}/teams/${teamId}/players/${playerUserId}/profile`, profileData);
  return data;
};

File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/eq-master/lib/services/player_service.dart
import '../models/api_models.dart';
import 'api_client.dart';

class PlayerService {
  final ApiClient _api = ApiClient.instance;

  Future<PlayerProfileDto> getMyProfile() async {
    final json = await _api.get('/players/me/profile');
    return PlayerProfileDto.fromJson(Map<String, dynamic>.from(json as Map));
  }

  Future<PlayerProfileDto> getPlayerProfile(
      String clubId, String teamId, String playerId) async {
    final json =
        await _api.get('/clubs/$clubId/teams/$teamId/players/$playerId/profile');
    return PlayerProfileDto.fromJson(Map<String, dynamic>.from(json as Map));
  }

  Future<PlayerProfileDto> upsertPlayerProfile(
      String clubId, String teamId, String playerId, Map<String, dynamic> body) async {
    final json = await _api.post(
      '/clubs/$clubId/teams/$teamId/players/$playerId/profile',
      body: body,
    );
    return PlayerProfileDto.fromJson(Map<String, dynamic>.from(json as Map));
  }
}
